### ISSUE TITLE
How to set custom error for a specific field and its rule

### ISSUE THREAD

**Issue by sudhir600** [Wixel/GUMP#250]
Hi Guys,
Gump is really well but I didn't find any way to set custom error message based on rule defined.

Example  from your repo - https://github.com/Wixel/GUMP/blob/master/examples/full_example.php

```
$rules = array(
	'username'    => 'required|alpha_numeric|max_len,100|min_len,6',
	'password'    => 'required|max_len,100|min_len,6',
	'email'       => 'required|valid_email',
	'gender'      => 'required|exact_len,1',
	'credit_card' => 'required|valid_cc',
	'bio'		  => 'required'
);
```

Here lets consider `username`, we have set 4 rules. So i wanted to set 4 own error message.
All message will be different and that will come from an external file.
So is there any way to achieve this. 

Yii2 does like this- 
`$model->addRule(['name', 'email'], 'required', ['message' => "Please provide {attribute}."])`
but this also not provide to set completly diffrent error message for both attribute and by rules (2X2)

**Comment by filisko** [comment]
Have a look to: https://github.com/Wixel/GUMP/releases/tag/v1.5.1

**Comment by sudhir600** [comment]
sorry for closing and reopened this issue. happend by mistake.

**Comment by sudhir600** [comment]
Hi @filisko, that nice, but in one of the scenario this setting this custom error is not enough. 
what i am doing is - 
```
$rules = array(
  'username'  => 'required|alpha_numeric|max_len,100|min_len,3',
  'phone'     => 'required|min_len,10|max_len,20',
);
```
See above code, i have set two min validation.
now what i want to set entirely diffrent  error for username and phone.
such as if username doen't meet min careteria, i will show, your username is invalid but when phone have not minimum length then i will show, "you mobile number should be 10 digit".

Here as per guideline if i do like this -

```
Yii::$app->GUMP->set_error_messages(array(
    "validate_min_len"  => "Dear user, your username should be 10 digit.. okey??",
));
```

now once you set this error for min, it will show everywhere. (any validation where you have use min validation).

Actually it should be input + validation wise, not only validation wise.
something like this- 

```
Array
(
    [userName] => Array
        (
            [required] => error msg for required
            [min] => error msg for required
            [max] => error msg for max
        )

    [phone] => Array
        (
            [min] => error msg for min but only for phone
        )

)
```

Hope you got my point.

**Comment by omzy83** [comment]
I am also looking for a solution to this. Laravel does it very well actually:

```
public function rules()
{
	return [
		'name' => 'required|string',
		'terms' => 'required|boolean',
	];
}

public function messages()
{
	return [
		'required' => 'This field is required',
		'terms.required' => 'The terms and conditions must be accepted',
	];
}
```

In the above, I can set a generic message for all "required" fields, and then a custom message for the "terms" required field.

**Comment by filisko** [comment]
In the meantime you can use $gump->get_errors_array() which returns field, value and rule that failed, then you can use your own helpers to create your error messages.

**Comment by filisko** [comment]
Released in 1.9.0
